pandas/core/generic.py

```diff
@@ -11326,20 +11326,30 @@ def cum_func(self, axis=None, skipna=True, *args, **kwargs):
         else:
             axis = self._get_axis_number(axis)
 
-        y = com.values_from_object(self).copy()
-
-        if skipna and issubclass(y.dtype.type, (np.datetime64, np.timedelta64)):
-            result = accum_func(y, axis)
-            mask = isna(self)
-            np.putmask(result, mask, iNaT)
-        elif skipna and not issubclass(y.dtype.type, (np.integer, np.bool_)):
-            mask = isna(self)
-            np.putmask(y, mask, mask_a)
-            result = accum_func(y, axis)
-            np.putmask(result, mask, mask_b)
-        else:
-            result = accum_func(y, axis)
+        if axis == 1:
+            return cum_func(self.T, axis=0, skipna=skipna, *args, **kwargs).T
```

Comment:
Is there a better way to do this than the double transpose?

Reply:
What do you have in mind?  Only having to worry about axis==0 is nice in part because we're going to work on block.values which can be 1D for EAs

Reply:
come to think of it, EAs are going to present more poorly-behaved corner cases in the axis=1 case since they aren't preserved under transpose (if only they were 2D!)

Reply:
https://github.com/pandas-dev/pandas/pull/30091 fixes that.

The double transpose will be painfully slow for EAs, so it'd be nice to avoid it if possible, but I'm not sure if it is.